packages/frontend/core/src/modules/app-sidebar/impls/storage.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  type GlobalState,
+  type Memento,
+  wrapMemento,
+} from '@toeverything/infra';
+
+import type { AppSidebarLocalState } from '../providers/storage';
+
+export class AppSidebarLocalStateImpl implements AppSidebarLocalState {
```

Comment:
the name is confusing. it is in fact a global state, not local.
could be renamed as `AppSidebarState` instead.